index.js

```diff
@@ -138,3 +138,67 @@ newSightingForm.addEventListener('submit', function (event) {
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
 
+
+// FIRST TRY //
+// const collectionUl = document.querySelector('ul#animals')
+
+// collectionUl.addEventListener('click', function (event){
+
+//     if (event.target.matches('button.like-button')){
+//         const likesPtag = event.target.previousElementSibling
+//         const likes = parseInt(likesPtag.textContent) + 1
+//         likesPtag.textContent = `${likes} Likes`
+//     }
+//     else if (event.target.matches('button.delete-button')){
+//         const animal = event.target.closest('li')
+//         animal.remove()
+//     }
+//     else if (event.target.matches('button.toggle-update-form-button')){
+//         debugger
+//         const animal = event.target.closest('#sighting-description')
+//         animal.style.display = "none"
+//     }
+// })
+
+
+// REVIEW: Different Way to Write the Code //
+
+const animalCollection = document.querySelector('ul#animals')
+
+animalCollection.addEventListener('click', function(event){
+
+    const li = event.target.closest('li')
+
+    if(event.target.matches('button.like-button')) {
+       
+        const likesPtag = li.querySelector('p.likes-display')
+        const currLikes = parseInt(likesPtag.textContent) + 1
+        likesPtag.textContent = `${currLikes} Likes`
+    }
+    else if(event.target.className === 'delete-button'){
+        const li = event.target.closest('li')
+        li.remove()
+    }
+    else if(event.target.matches('button.toggle-update-form-button')){
+        //we have to define the form
+        const form = li.querySelector('form')
+        const updateForm = event.target.nextElementSibling
+        if(form.style.display === 'block'){
+            form.style.display = 'none'
+        }
+        else {
+            form.style.display = 'block'
+        }
+        //form.style.display = form.style.display === 'block' ? 'none' : 'block'
+    }
+})
+
+animalCollection.addEventListener('submit', function (event){
+    if (event.target.matches('form.update-form')){
+        event.preventDefault()
+        const li = event.target.closest('li')
+        const descriptionInput = event.target[0].value
+        const descriptionPtag = li.querySelector('p')
+        descriptionPtag.textContent = descriptionInput
+    }
+})
```

Comment:
Great work here as well. You are listening for any form submission that occurs within the animals UL element. Then you make sure to prevent the default action of the form. Then you find the p tag and update its value based on the form data. Don't forget to change the form field value to the new description. Again, we can break it apart into 2 steps.

```
/***** BONUS DELIVERABLE part 1 *****/

animalsUL.addEventListener("submit", function(event){
    event.preventDefault()
    if (event.target.matches('.update-form')){
        handleUpdateFormSubmission(event)
    }
})

/***** BONUS DELIVERABLE part 2 *****/
function handleUpdateFormSubmission(e){
    e.preventDefault()
    const description = e.target.children[0].value
    const animalDescriptionPTag = e.target.parentElement.children[0]
    animalDescriptionPTag.innerText = description
    e.target.value = description
}
```